### PR TITLE
Support for setting user in default executor

### LINKED ISSUE #1261: select user to launch tasks as
Is it possible to select the user used to execute a task as? I am thinking something similar to supervisor's user option in the processes definition which ultimately uses os.setgroups/os.setgid/os.setuid. Thanks.

### PR DESCRIPTION
**Pull request by ssalinas**
/fixes #1261

### CODE REVIEW

**Comment by kchaliki** [comment]
@ssalinas thanks for the addition

File: SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosTaskBuilder.java [hunk 1]
```diff
@@ -381,6 +381,10 @@ private void prepareCustomExecutor(final TaskInfo.Builder bldr, final Singularit
   private void prepareCommand(final TaskInfo.Builder bldr, final SingularityTaskId taskId, final SingularityTaskRequest task, final Protos.Offer offer, final Optional<long[]> ports) {
     CommandInfo.Builder commandBldr = CommandInfo.newBuilder();
 
+    if (task.getDeploy().getUser().isPresent()) {
+      commandBldr.setUser(task.getDeploy().getUser().get());
```
Comment (tpetr) [inline SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosTaskBuilder.java]: Can we add this corresponding change to `prepareCustomExecutor()` too? I can see some utility in being able to control what user the custom executor is launched as.

**Comment by tpetr** [comment]
LGTM aside from comment above

File: SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosTaskBuilder.java (hunk 1 above)
Comment (ssalinas) [inline SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosTaskBuilder.java]: We already had `customExecutorUser` for this purpose, but it had been deprecated for a while now. I've officially removed it in favor of the plain `user` field which will apply to the custom and default executor cases.

**Pull request merged**